catroid/src/main/java/org/catrobat/catroid/utils/CrashReporter.java

```diff
@@ -96,29 +88,15 @@ public static boolean logException(Throwable t) {
 		return false;
 	}
 
-	public static boolean checkIfCrashRecoveryAndFinishActivity(final Activity context) {
-		Log.d(TAG, "AFTER_EXCEPTION : checkIfCrashRecoveryAndFinishActivity()");
-		if (isRecoveredFromCrash()) {
-			if (isReportingEnabled()) {
-				sendUnhandledCaughtException();
-			}
-			if (!(context instanceof MainMenuActivity)) {
-				context.finish();
-			} else {
-				preferences.edit().putBoolean(RECOVERED_FROM_CRASH, false).commit();
-				return true;
-			}
-		}
-		return false;
-	}
-
 	public static void sendUnhandledCaughtException() {
-		Log.d(TAG, "AFTER_EXCEPTION : sendCaughtException()");
-		Gson gson = new Gson();
 		String json = preferences.getString(EXCEPTION_FOR_REPORT, "");
-		Throwable exception = gson.fromJson(json, Throwable.class);
-		if (logException(exception)) {
-			preferences.edit().remove(EXCEPTION_FOR_REPORT).commit();
+		if (isReportingEnabled() && !json.isEmpty()) {
+			Log.d(TAG, "AFTER_EXCEPTION : sendCaughtException()");
+			Gson gson = new Gson();
+			Throwable exception = gson.fromJson(json, Throwable.class);
+			if (logException(exception)) {
```

Comment:
please make logException return void.
- the boolean return value is only utilized here.
- its always going to be true, since in line 93  you made sure that its only reached when isReportingEnabled() is true